lua/plenary/path.lua

```diff
@@ -296,7 +296,11 @@ function Path:_st_mode()
 end
 
 function Path:joinpath(...)
-  return Path:new(self.filename, ...)
+  if string.char(self.filename:byte(-1)) == path.sep then
+    return Path:new(self.filename:sub(0, -2), ...)
+  else
+    return Path:new(self.filename, ...)
+  end
```

Comment:
Maybe it's better to split these change out to it's own separate PR?

Reply:
That's a good point. I think this implementation is better, but still not actually correct. It deserves it's own thread.